{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'tidyverse' was built under R version 3.6.2\"-- Attaching packages --------------------------------------- tidyverse 1.3.0 --\n",
      "v ggplot2 3.3.0     v purrr   0.3.3\n",
      "v tibble  2.1.3     v dplyr   0.8.4\n",
      "v tidyr   1.0.2     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.4.0\n",
      "Warning message:\n",
      "\"package 'ggplot2' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'tibble' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'tidyr' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'readr' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'purrr' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'dplyr' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'stringr' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'forcats' was built under R version 3.6.2\"-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::filter() masks stats::filter()\n",
      "x dplyr::lag()    masks stats::lag()\n",
      "Warning message:\n",
      "\"package 'nycflights13' was built under R version 3.6.2\"Warning message:\n",
      "\"package 'gridExtra' was built under R version 3.6.2\"\n",
      "Attaching package: 'gridExtra'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    combine\n",
      "\n",
      "Warning message:\n",
      "\"package 'reshape' was built under R version 3.6.2\"\n",
      "Attaching package: 'reshape'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    rename\n",
      "\n",
      "The following objects are masked from 'package:tidyr':\n",
      "\n",
      "    expand, smiths\n",
      "\n",
      "Warning message:\n",
      "\"package 'edgeR' was built under R version 3.6.2\"Loading required package: limma\n",
      "Warning message:\n",
      "\"package 'limma' was built under R version 3.6.2\"Loading required package: S4Vectors\n",
      "Warning message:\n",
      "\"package 'S4Vectors' was built under R version 3.6.2\"Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following object is masked from 'package:limma':\n",
      "\n",
      "    plotMA\n",
      "\n",
      "The following object is masked from 'package:gridExtra':\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "The following objects are masked from 'package:reshape':\n",
      "\n",
      "    expand, rename\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    first, rename\n",
      "\n",
      "The following object is masked from 'package:tidyr':\n",
      "\n",
      "    expand\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Warning message:\n",
      "\"package 'IRanges' was built under R version 3.6.2\"\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "The following object is masked from 'package:purrr':\n",
      "\n",
      "    reduce\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: SummarizedExperiment\n",
      "Warning message:\n",
      "\"package 'SummarizedExperiment' was built under R version 3.6.2\"Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "Warning message:\n",
      "\"package 'DelayedArray' was built under R version 3.6.2\"Loading required package: matrixStats\n",
      "Warning message:\n",
      "\"package 'matrixStats' was built under R version 3.6.3\"\n",
      "Attaching package: 'matrixStats'\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    count\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "Warning message:\n",
      "\"package 'BiocParallel' was built under R version 3.6.2\"\n",
      "Attaching package: 'DelayedArray'\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "The following object is masked from 'package:purrr':\n",
      "\n",
      "    simplify\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "Warning message:\n",
      "\"package 'fpc' was built under R version 3.6.3\"\n",
      "Attaching package: 'MASS'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    select\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(nycflights13)\n",
    "library(ggplot2)\n",
    "library(gridExtra)\n",
    "library(grid)\n",
    "library(reshape)\n",
    "library(stringr)\n",
    "library(edgeR)\n",
    "library(cluster)\n",
    "library(DESeq2)\n",
    "library(limma)\n",
    "library(fpc)\n",
    "library(MASS)\n",
    "library(scatterplot3d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical <- read.csv(\"clinical_data_breast_cancer.csv\")\n",
    "itraq <- read.csv(\"77_cancer_proteomes_CPTAC_itraq.csv\")\n",
    "pam50 <- read.csv(\"PAM50_proteins.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Complete.TCGA.ID'</li>\n",
       "\t<li>'Gender'</li>\n",
       "\t<li>'Age.at.Initial.Pathologic.Diagnosis'</li>\n",
       "\t<li>'ER.Status'</li>\n",
       "\t<li>'PR.Status'</li>\n",
       "\t<li>'HER2.Final.Status'</li>\n",
       "\t<li>'Tumor'</li>\n",
       "\t<li>'Tumor..T1.Coded'</li>\n",
       "\t<li>'Node'</li>\n",
       "\t<li>'Node.Coded'</li>\n",
       "\t<li>'Metastasis'</li>\n",
       "\t<li>'Metastasis.Coded'</li>\n",
       "\t<li>'AJCC.Stage'</li>\n",
       "\t<li>'Converted.Stage'</li>\n",
       "\t<li>'Survival.Data.Form'</li>\n",
       "\t<li>'Vital.Status'</li>\n",
       "\t<li>'Days.to.Date.of.Last.Contact'</li>\n",
       "\t<li>'Days.to.date.of.Death'</li>\n",
       "\t<li>'OS.event'</li>\n",
       "\t<li>'OS.Time'</li>\n",
       "\t<li>'PAM50.mRNA'</li>\n",
       "\t<li>'SigClust.Unsupervised.mRNA'</li>\n",
       "\t<li>'SigClust.Intrinsic.mRNA'</li>\n",
       "\t<li>'miRNA.Clusters'</li>\n",
       "\t<li>'methylation.Clusters'</li>\n",
       "\t<li>'RPPA.Clusters'</li>\n",
       "\t<li>'CN.Clusters'</li>\n",
       "\t<li>'Integrated.Clusters..with.PAM50.'</li>\n",
       "\t<li>'Integrated.Clusters..no.exp.'</li>\n",
       "\t<li>'Integrated.Clusters..unsup.exp.'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Complete.TCGA.ID'\n",
       "\\item 'Gender'\n",
       "\\item 'Age.at.Initial.Pathologic.Diagnosis'\n",
       "\\item 'ER.Status'\n",
       "\\item 'PR.Status'\n",
       "\\item 'HER2.Final.Status'\n",
       "\\item 'Tumor'\n",
       "\\item 'Tumor..T1.Coded'\n",
       "\\item 'Node'\n",
       "\\item 'Node.Coded'\n",
       "\\item 'Metastasis'\n",
       "\\item 'Metastasis.Coded'\n",
       "\\item 'AJCC.Stage'\n",
       "\\item 'Converted.Stage'\n",
       "\\item 'Survival.Data.Form'\n",
       "\\item 'Vital.Status'\n",
       "\\item 'Days.to.Date.of.Last.Contact'\n",
       "\\item 'Days.to.date.of.Death'\n",
       "\\item 'OS.event'\n",
       "\\item 'OS.Time'\n",
       "\\item 'PAM50.mRNA'\n",
       "\\item 'SigClust.Unsupervised.mRNA'\n",
       "\\item 'SigClust.Intrinsic.mRNA'\n",
       "\\item 'miRNA.Clusters'\n",
       "\\item 'methylation.Clusters'\n",
       "\\item 'RPPA.Clusters'\n",
       "\\item 'CN.Clusters'\n",
       "\\item 'Integrated.Clusters..with.PAM50.'\n",
       "\\item 'Integrated.Clusters..no.exp.'\n",
       "\\item 'Integrated.Clusters..unsup.exp.'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Complete.TCGA.ID'\n",
       "2. 'Gender'\n",
       "3. 'Age.at.Initial.Pathologic.Diagnosis'\n",
       "4. 'ER.Status'\n",
       "5. 'PR.Status'\n",
       "6. 'HER2.Final.Status'\n",
       "7. 'Tumor'\n",
       "8. 'Tumor..T1.Coded'\n",
       "9. 'Node'\n",
       "10. 'Node.Coded'\n",
       "11. 'Metastasis'\n",
       "12. 'Metastasis.Coded'\n",
       "13. 'AJCC.Stage'\n",
       "14. 'Converted.Stage'\n",
       "15. 'Survival.Data.Form'\n",
       "16. 'Vital.Status'\n",
       "17. 'Days.to.Date.of.Last.Contact'\n",
       "18. 'Days.to.date.of.Death'\n",
       "19. 'OS.event'\n",
       "20. 'OS.Time'\n",
       "21. 'PAM50.mRNA'\n",
       "22. 'SigClust.Unsupervised.mRNA'\n",
       "23. 'SigClust.Intrinsic.mRNA'\n",
       "24. 'miRNA.Clusters'\n",
       "25. 'methylation.Clusters'\n",
       "26. 'RPPA.Clusters'\n",
       "27. 'CN.Clusters'\n",
       "28. 'Integrated.Clusters..with.PAM50.'\n",
       "29. 'Integrated.Clusters..no.exp.'\n",
       "30. 'Integrated.Clusters..unsup.exp.'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Complete.TCGA.ID\"                    \"Gender\"                             \n",
       " [3] \"Age.at.Initial.Pathologic.Diagnosis\" \"ER.Status\"                          \n",
       " [5] \"PR.Status\"                           \"HER2.Final.Status\"                  \n",
       " [7] \"Tumor\"                               \"Tumor..T1.Coded\"                    \n",
       " [9] \"Node\"                                \"Node.Coded\"                         \n",
       "[11] \"Metastasis\"                          \"Metastasis.Coded\"                   \n",
       "[13] \"AJCC.Stage\"                          \"Converted.Stage\"                    \n",
       "[15] \"Survival.Data.Form\"                  \"Vital.Status\"                       \n",
       "[17] \"Days.to.Date.of.Last.Contact\"        \"Days.to.date.of.Death\"              \n",
       "[19] \"OS.event\"                            \"OS.Time\"                            \n",
       "[21] \"PAM50.mRNA\"                          \"SigClust.Unsupervised.mRNA\"         \n",
       "[23] \"SigClust.Intrinsic.mRNA\"             \"miRNA.Clusters\"                     \n",
       "[25] \"methylation.Clusters\"                \"RPPA.Clusters\"                      \n",
       "[27] \"CN.Clusters\"                         \"Integrated.Clusters..with.PAM50.\"   \n",
       "[29] \"Integrated.Clusters..no.exp.\"        \"Integrated.Clusters..unsup.exp.\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>105</li>\n",
       "\t<li>30</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 105\n",
       "\\item 30\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 105\n",
       "2. 30\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 105  30"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'RefSeq_accession_number'</li>\n",
       "\t<li>'gene_symbol'</li>\n",
       "\t<li>'gene_name'</li>\n",
       "\t<li>'AO.A12D.01TCGA'</li>\n",
       "\t<li>'C8.A131.01TCGA'</li>\n",
       "\t<li>'AO.A12B.01TCGA'</li>\n",
       "\t<li>'BH.A18Q.02TCGA'</li>\n",
       "\t<li>'C8.A130.02TCGA'</li>\n",
       "\t<li>'C8.A138.03TCGA'</li>\n",
       "\t<li>'E2.A154.03TCGA'</li>\n",
       "\t<li>'C8.A12L.04TCGA'</li>\n",
       "\t<li>'A2.A0EX.04TCGA'</li>\n",
       "\t<li>'AO.A12D.05TCGA'</li>\n",
       "\t<li>'AN.A04A.05TCGA'</li>\n",
       "\t<li>'BH.A0AV.05TCGA'</li>\n",
       "\t<li>'C8.A12T.06TCGA'</li>\n",
       "\t<li>'A8.A06Z.07TCGA'</li>\n",
       "\t<li>'A2.A0CM.07TCGA'</li>\n",
       "\t<li>'BH.A18U.08TCGA'</li>\n",
       "\t<li>'A2.A0EQ.08TCGA'</li>\n",
       "\t<li>'AR.A0U4.09TCGA'</li>\n",
       "\t<li>'AO.A0J9.10TCGA'</li>\n",
       "\t<li>'AR.A1AP.11TCGA'</li>\n",
       "\t<li>'AN.A0FK.11TCGA'</li>\n",
       "\t<li>'AO.A0J6.11TCGA'</li>\n",
       "\t<li>'A7.A13F.12TCGA'</li>\n",
       "\t<li>'BH.A0E1.12TCGA'</li>\n",
       "\t<li>'A7.A0CE.13TCGA'</li>\n",
       "\t<li>'A2.A0YC.13TCGA'</li>\n",
       "\t<li>'AO.A0JC.14TCGA'</li>\n",
       "\t<li>'A8.A08Z.14TCGA'</li>\n",
       "\t<li>'AR.A0TX.14TCGA'</li>\n",
       "\t<li>'A8.A076.15TCGA'</li>\n",
       "\t<li>'AO.A126.15TCGA'</li>\n",
       "\t<li>'BH.A0C1.16TCGA'</li>\n",
       "\t<li>'A2.A0EY.16TCGA'</li>\n",
       "\t<li>'AR.A1AW.17TCGA'</li>\n",
       "\t<li>'AR.A1AV.17TCGA'</li>\n",
       "\t<li>'C8.A135.17TCGA'</li>\n",
       "\t<li>'A2.A0EV.18TCGA'</li>\n",
       "\t<li>'AN.A0AM.18TCGA'</li>\n",
       "\t<li>'D8.A142.18TCGA'</li>\n",
       "\t<li>'AN.A0FL.19TCGA'</li>\n",
       "\t<li>'BH.A0DG.19TCGA'</li>\n",
       "\t<li>'AR.A0TV.20TCGA'</li>\n",
       "\t<li>'C8.A12Z.20TCGA'</li>\n",
       "\t<li>'AO.A0JJ.20TCGA'</li>\n",
       "\t<li>'AO.A0JE.21TCGA'</li>\n",
       "\t<li>'AN.A0AJ.21TCGA'</li>\n",
       "\t<li>'A7.A0CJ.22TCGA'</li>\n",
       "\t<li>'AO.A12F.22TCGA'</li>\n",
       "\t<li>'A8.A079.23TCGA'</li>\n",
       "\t<li>'A2.A0T3.24TCGA'</li>\n",
       "\t<li>'A2.A0YD.24TCGA'</li>\n",
       "\t<li>'AR.A0TR.25TCGA'</li>\n",
       "\t<li>'AO.A03O.25TCGA'</li>\n",
       "\t<li>'AO.A12E.26TCGA'</li>\n",
       "\t<li>'A8.A06N.26TCGA'</li>\n",
       "\t<li>'A2.A0YG.27TCGA'</li>\n",
       "\t<li>'BH.A18N.27TCGA'</li>\n",
       "\t<li>'AN.A0AL.28TCGA'</li>\n",
       "\t<li>'A2.A0T6.29TCGA'</li>\n",
       "\t<li>'E2.A158.29TCGA'</li>\n",
       "\t<li>'E2.A15A.29TCGA'</li>\n",
       "\t<li>'AO.A0JM.30TCGA'</li>\n",
       "\t<li>'C8.A12V.30TCGA'</li>\n",
       "\t<li>'A2.A0D2.31TCGA'</li>\n",
       "\t<li>'C8.A12U.31TCGA'</li>\n",
       "\t<li>'AR.A1AS.31TCGA'</li>\n",
       "\t<li>'A8.A09G.32TCGA'</li>\n",
       "\t<li>'C8.A131.32TCGA'</li>\n",
       "\t<li>'C8.A134.32TCGA'</li>\n",
       "\t<li>'A2.A0YF.33TCGA'</li>\n",
       "\t<li>'BH.A0DD.33TCGA'</li>\n",
       "\t<li>'BH.A0E9.33TCGA'</li>\n",
       "\t<li>'AR.A0TT.34TCGA'</li>\n",
       "\t<li>'AO.A12B.34TCGA'</li>\n",
       "\t<li>'A2.A0SW.35TCGA'</li>\n",
       "\t<li>'AO.A0JL.35TCGA'</li>\n",
       "\t<li>'BH.A0BV.35TCGA'</li>\n",
       "\t<li>'A2.A0YM.36TCGA'</li>\n",
       "\t<li>'BH.A0C7.36TCGA'</li>\n",
       "\t<li>'A2.A0SX.36TCGA'</li>\n",
       "\t<li>'X263d3f.I.CPTAC'</li>\n",
       "\t<li>'blcdb9.I.CPTAC'</li>\n",
       "\t<li>'c4155b.C.CPTAC'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'RefSeq\\_accession\\_number'\n",
       "\\item 'gene\\_symbol'\n",
       "\\item 'gene\\_name'\n",
       "\\item 'AO.A12D.01TCGA'\n",
       "\\item 'C8.A131.01TCGA'\n",
       "\\item 'AO.A12B.01TCGA'\n",
       "\\item 'BH.A18Q.02TCGA'\n",
       "\\item 'C8.A130.02TCGA'\n",
       "\\item 'C8.A138.03TCGA'\n",
       "\\item 'E2.A154.03TCGA'\n",
       "\\item 'C8.A12L.04TCGA'\n",
       "\\item 'A2.A0EX.04TCGA'\n",
       "\\item 'AO.A12D.05TCGA'\n",
       "\\item 'AN.A04A.05TCGA'\n",
       "\\item 'BH.A0AV.05TCGA'\n",
       "\\item 'C8.A12T.06TCGA'\n",
       "\\item 'A8.A06Z.07TCGA'\n",
       "\\item 'A2.A0CM.07TCGA'\n",
       "\\item 'BH.A18U.08TCGA'\n",
       "\\item 'A2.A0EQ.08TCGA'\n",
       "\\item 'AR.A0U4.09TCGA'\n",
       "\\item 'AO.A0J9.10TCGA'\n",
       "\\item 'AR.A1AP.11TCGA'\n",
       "\\item 'AN.A0FK.11TCGA'\n",
       "\\item 'AO.A0J6.11TCGA'\n",
       "\\item 'A7.A13F.12TCGA'\n",
       "\\item 'BH.A0E1.12TCGA'\n",
       "\\item 'A7.A0CE.13TCGA'\n",
       "\\item 'A2.A0YC.13TCGA'\n",
       "\\item 'AO.A0JC.14TCGA'\n",
       "\\item 'A8.A08Z.14TCGA'\n",
       "\\item 'AR.A0TX.14TCGA'\n",
       "\\item 'A8.A076.15TCGA'\n",
       "\\item 'AO.A126.15TCGA'\n",
       "\\item 'BH.A0C1.16TCGA'\n",
       "\\item 'A2.A0EY.16TCGA'\n",
       "\\item 'AR.A1AW.17TCGA'\n",
       "\\item 'AR.A1AV.17TCGA'\n",
       "\\item 'C8.A135.17TCGA'\n",
       "\\item 'A2.A0EV.18TCGA'\n",
       "\\item 'AN.A0AM.18TCGA'\n",
       "\\item 'D8.A142.18TCGA'\n",
       "\\item 'AN.A0FL.19TCGA'\n",
       "\\item 'BH.A0DG.19TCGA'\n",
       "\\item 'AR.A0TV.20TCGA'\n",
       "\\item 'C8.A12Z.20TCGA'\n",
       "\\item 'AO.A0JJ.20TCGA'\n",
       "\\item 'AO.A0JE.21TCGA'\n",
       "\\item 'AN.A0AJ.21TCGA'\n",
       "\\item 'A7.A0CJ.22TCGA'\n",
       "\\item 'AO.A12F.22TCGA'\n",
       "\\item 'A8.A079.23TCGA'\n",
       "\\item 'A2.A0T3.24TCGA'\n",
       "\\item 'A2.A0YD.24TCGA'\n",
       "\\item 'AR.A0TR.25TCGA'\n",
       "\\item 'AO.A03O.25TCGA'\n",
       "\\item 'AO.A12E.26TCGA'\n",
       "\\item 'A8.A06N.26TCGA'\n",
       "\\item 'A2.A0YG.27TCGA'\n",
       "\\item 'BH.A18N.27TCGA'\n",
       "\\item 'AN.A0AL.28TCGA'\n",
       "\\item 'A2.A0T6.29TCGA'\n",
       "\\item 'E2.A158.29TCGA'\n",
       "\\item 'E2.A15A.29TCGA'\n",
       "\\item 'AO.A0JM.30TCGA'\n",
       "\\item 'C8.A12V.30TCGA'\n",
       "\\item 'A2.A0D2.31TCGA'\n",
       "\\item 'C8.A12U.31TCGA'\n",
       "\\item 'AR.A1AS.31TCGA'\n",
       "\\item 'A8.A09G.32TCGA'\n",
       "\\item 'C8.A131.32TCGA'\n",
       "\\item 'C8.A134.32TCGA'\n",
       "\\item 'A2.A0YF.33TCGA'\n",
       "\\item 'BH.A0DD.33TCGA'\n",
       "\\item 'BH.A0E9.33TCGA'\n",
       "\\item 'AR.A0TT.34TCGA'\n",
       "\\item 'AO.A12B.34TCGA'\n",
       "\\item 'A2.A0SW.35TCGA'\n",
       "\\item 'AO.A0JL.35TCGA'\n",
       "\\item 'BH.A0BV.35TCGA'\n",
       "\\item 'A2.A0YM.36TCGA'\n",
       "\\item 'BH.A0C7.36TCGA'\n",
       "\\item 'A2.A0SX.36TCGA'\n",
       "\\item 'X263d3f.I.CPTAC'\n",
       "\\item 'blcdb9.I.CPTAC'\n",
       "\\item 'c4155b.C.CPTAC'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'RefSeq_accession_number'\n",
       "2. 'gene_symbol'\n",
       "3. 'gene_name'\n",
       "4. 'AO.A12D.01TCGA'\n",
       "5. 'C8.A131.01TCGA'\n",
       "6. 'AO.A12B.01TCGA'\n",
       "7. 'BH.A18Q.02TCGA'\n",
       "8. 'C8.A130.02TCGA'\n",
       "9. 'C8.A138.03TCGA'\n",
       "10. 'E2.A154.03TCGA'\n",
       "11. 'C8.A12L.04TCGA'\n",
       "12. 'A2.A0EX.04TCGA'\n",
       "13. 'AO.A12D.05TCGA'\n",
       "14. 'AN.A04A.05TCGA'\n",
       "15. 'BH.A0AV.05TCGA'\n",
       "16. 'C8.A12T.06TCGA'\n",
       "17. 'A8.A06Z.07TCGA'\n",
       "18. 'A2.A0CM.07TCGA'\n",
       "19. 'BH.A18U.08TCGA'\n",
       "20. 'A2.A0EQ.08TCGA'\n",
       "21. 'AR.A0U4.09TCGA'\n",
       "22. 'AO.A0J9.10TCGA'\n",
       "23. 'AR.A1AP.11TCGA'\n",
       "24. 'AN.A0FK.11TCGA'\n",
       "25. 'AO.A0J6.11TCGA'\n",
       "26. 'A7.A13F.12TCGA'\n",
       "27. 'BH.A0E1.12TCGA'\n",
       "28. 'A7.A0CE.13TCGA'\n",
       "29. 'A2.A0YC.13TCGA'\n",
       "30. 'AO.A0JC.14TCGA'\n",
       "31. 'A8.A08Z.14TCGA'\n",
       "32. 'AR.A0TX.14TCGA'\n",
       "33. 'A8.A076.15TCGA'\n",
       "34. 'AO.A126.15TCGA'\n",
       "35. 'BH.A0C1.16TCGA'\n",
       "36. 'A2.A0EY.16TCGA'\n",
       "37. 'AR.A1AW.17TCGA'\n",
       "38. 'AR.A1AV.17TCGA'\n",
       "39. 'C8.A135.17TCGA'\n",
       "40. 'A2.A0EV.18TCGA'\n",
       "41. 'AN.A0AM.18TCGA'\n",
       "42. 'D8.A142.18TCGA'\n",
       "43. 'AN.A0FL.19TCGA'\n",
       "44. 'BH.A0DG.19TCGA'\n",
       "45. 'AR.A0TV.20TCGA'\n",
       "46. 'C8.A12Z.20TCGA'\n",
       "47. 'AO.A0JJ.20TCGA'\n",
       "48. 'AO.A0JE.21TCGA'\n",
       "49. 'AN.A0AJ.21TCGA'\n",
       "50. 'A7.A0CJ.22TCGA'\n",
       "51. 'AO.A12F.22TCGA'\n",
       "52. 'A8.A079.23TCGA'\n",
       "53. 'A2.A0T3.24TCGA'\n",
       "54. 'A2.A0YD.24TCGA'\n",
       "55. 'AR.A0TR.25TCGA'\n",
       "56. 'AO.A03O.25TCGA'\n",
       "57. 'AO.A12E.26TCGA'\n",
       "58. 'A8.A06N.26TCGA'\n",
       "59. 'A2.A0YG.27TCGA'\n",
       "60. 'BH.A18N.27TCGA'\n",
       "61. 'AN.A0AL.28TCGA'\n",
       "62. 'A2.A0T6.29TCGA'\n",
       "63. 'E2.A158.29TCGA'\n",
       "64. 'E2.A15A.29TCGA'\n",
       "65. 'AO.A0JM.30TCGA'\n",
       "66. 'C8.A12V.30TCGA'\n",
       "67. 'A2.A0D2.31TCGA'\n",
       "68. 'C8.A12U.31TCGA'\n",
       "69. 'AR.A1AS.31TCGA'\n",
       "70. 'A8.A09G.32TCGA'\n",
       "71. 'C8.A131.32TCGA'\n",
       "72. 'C8.A134.32TCGA'\n",
       "73. 'A2.A0YF.33TCGA'\n",
       "74. 'BH.A0DD.33TCGA'\n",
       "75. 'BH.A0E9.33TCGA'\n",
       "76. 'AR.A0TT.34TCGA'\n",
       "77. 'AO.A12B.34TCGA'\n",
       "78. 'A2.A0SW.35TCGA'\n",
       "79. 'AO.A0JL.35TCGA'\n",
       "80. 'BH.A0BV.35TCGA'\n",
       "81. 'A2.A0YM.36TCGA'\n",
       "82. 'BH.A0C7.36TCGA'\n",
       "83. 'A2.A0SX.36TCGA'\n",
       "84. 'X263d3f.I.CPTAC'\n",
       "85. 'blcdb9.I.CPTAC'\n",
       "86. 'c4155b.C.CPTAC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"RefSeq_accession_number\" \"gene_symbol\"            \n",
       " [3] \"gene_name\"               \"AO.A12D.01TCGA\"         \n",
       " [5] \"C8.A131.01TCGA\"          \"AO.A12B.01TCGA\"         \n",
       " [7] \"BH.A18Q.02TCGA\"          \"C8.A130.02TCGA\"         \n",
       " [9] \"C8.A138.03TCGA\"          \"E2.A154.03TCGA\"         \n",
       "[11] \"C8.A12L.04TCGA\"          \"A2.A0EX.04TCGA\"         \n",
       "[13] \"AO.A12D.05TCGA\"          \"AN.A04A.05TCGA\"         \n",
       "[15] \"BH.A0AV.05TCGA\"          \"C8.A12T.06TCGA\"         \n",
       "[17] \"A8.A06Z.07TCGA\"          \"A2.A0CM.07TCGA\"         \n",
       "[19] \"BH.A18U.08TCGA\"          \"A2.A0EQ.08TCGA\"         \n",
       "[21] \"AR.A0U4.09TCGA\"          \"AO.A0J9.10TCGA\"         \n",
       "[23] \"AR.A1AP.11TCGA\"          \"AN.A0FK.11TCGA\"         \n",
       "[25] \"AO.A0J6.11TCGA\"          \"A7.A13F.12TCGA\"         \n",
       "[27] \"BH.A0E1.12TCGA\"          \"A7.A0CE.13TCGA\"         \n",
       "[29] \"A2.A0YC.13TCGA\"          \"AO.A0JC.14TCGA\"         \n",
       "[31] \"A8.A08Z.14TCGA\"          \"AR.A0TX.14TCGA\"         \n",
       "[33] \"A8.A076.15TCGA\"          \"AO.A126.15TCGA\"         \n",
       "[35] \"BH.A0C1.16TCGA\"          \"A2.A0EY.16TCGA\"         \n",
       "[37] \"AR.A1AW.17TCGA\"          \"AR.A1AV.17TCGA\"         \n",
       "[39] \"C8.A135.17TCGA\"          \"A2.A0EV.18TCGA\"         \n",
       "[41] \"AN.A0AM.18TCGA\"          \"D8.A142.18TCGA\"         \n",
       "[43] \"AN.A0FL.19TCGA\"          \"BH.A0DG.19TCGA\"         \n",
       "[45] \"AR.A0TV.20TCGA\"          \"C8.A12Z.20TCGA\"         \n",
       "[47] \"AO.A0JJ.20TCGA\"          \"AO.A0JE.21TCGA\"         \n",
       "[49] \"AN.A0AJ.21TCGA\"          \"A7.A0CJ.22TCGA\"         \n",
       "[51] \"AO.A12F.22TCGA\"          \"A8.A079.23TCGA\"         \n",
       "[53] \"A2.A0T3.24TCGA\"          \"A2.A0YD.24TCGA\"         \n",
       "[55] \"AR.A0TR.25TCGA\"          \"AO.A03O.25TCGA\"         \n",
       "[57] \"AO.A12E.26TCGA\"          \"A8.A06N.26TCGA\"         \n",
       "[59] \"A2.A0YG.27TCGA\"          \"BH.A18N.27TCGA\"         \n",
       "[61] \"AN.A0AL.28TCGA\"          \"A2.A0T6.29TCGA\"         \n",
       "[63] \"E2.A158.29TCGA\"          \"E2.A15A.29TCGA\"         \n",
       "[65] \"AO.A0JM.30TCGA\"          \"C8.A12V.30TCGA\"         \n",
       "[67] \"A2.A0D2.31TCGA\"          \"C8.A12U.31TCGA\"         \n",
       "[69] \"AR.A1AS.31TCGA\"          \"A8.A09G.32TCGA\"         \n",
       "[71] \"C8.A131.32TCGA\"          \"C8.A134.32TCGA\"         \n",
       "[73] \"A2.A0YF.33TCGA\"          \"BH.A0DD.33TCGA\"         \n",
       "[75] \"BH.A0E9.33TCGA\"          \"AR.A0TT.34TCGA\"         \n",
       "[77] \"AO.A12B.34TCGA\"          \"A2.A0SW.35TCGA\"         \n",
       "[79] \"AO.A0JL.35TCGA\"          \"BH.A0BV.35TCGA\"         \n",
       "[81] \"A2.A0YM.36TCGA\"          \"BH.A0C7.36TCGA\"         \n",
       "[83] \"A2.A0SX.36TCGA\"          \"X263d3f.I.CPTAC\"        \n",
       "[85] \"blcdb9.I.CPTAC\"          \"c4155b.C.CPTAC\"         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>12553</li>\n",
       "\t<li>86</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 12553\n",
       "\\item 86\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 12553\n",
       "2. 86\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 12553    86"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'gene_symbol'</li>\n",
       "\t<li>'RefSeqProteinID'</li>\n",
       "\t<li>'Species'</li>\n",
       "\t<li>'Gene.Name'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'gene\\_symbol'\n",
       "\\item 'RefSeqProteinID'\n",
       "\\item 'Species'\n",
       "\\item 'Gene.Name'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'gene_symbol'\n",
       "2. 'RefSeqProteinID'\n",
       "3. 'Species'\n",
       "4. 'Gene.Name'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"gene_symbol\"     \"RefSeqProteinID\" \"Species\"         \"Gene.Name\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>100</li>\n",
       "\t<li>4</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 100\n",
       "\\item 4\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 100\n",
       "2. 4\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 100   4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(clinical)\n",
    "dim(clinical)\n",
    "colnames(itraq)\n",
    "dim(itraq)\n",
    "colnames(pam50)\n",
    "dim(pam50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>12621</li>\n",
       "\t<li>89</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 12621\n",
       "\\item 89\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 12621\n",
       "2. 89\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 12621    89"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(pam50)[1] = colnames(itraq)[2]\n",
    "metadata <- merge(x=itraq,y=pam50,by=\"gene_symbol\",all=TRUE)\n",
    "dim(metadata)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
